data-prepper-api/src/main/java/org/opensearch/dataprepper/model/sink/AbstractSink.java

```diff
@@ -30,8 +30,12 @@ public abstract class AbstractSink<T extends Record<?>> implements Sink<T> {
     private int waitTimeMs;
     private SinkThread sinkThread;
 
+    protected String getName(final PluginSetting pluginSetting) {
```

Comment:
I don't think we should change `AbstractSink` for renaming. Rather, let the concrete sink pass in the `PluginMetrics`. Then handle the renaming in the `LambdaSink` directly.


```
super(pluginSetting, PluginMettrics.fromPluginSetting(pluginSetting));
```